src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -114,141 +114,145 @@ private static Task StartProcessWithKeepAlive(ProcessMetadata metadata)
             var path = metadata.ProcessPath;
             Log.Debug("Starting keep alive for {Process}.", path);
 
-            return Task.Run(
-                async () =>
+            return Task.Factory.StartNew(state => InternalStartProcessWithKeepAlive(state), metadata);
```

Comment:
Should this use `TaskCreationOptions.LongRunning`? Similar to https://github.com/DataDog/dogstatsd-csharp-client/pull/151.

Reply:
I guess it should. I was removing the closure, but you’re right.

Reply:
The method is async, so there's no benefit in using LongRunning (the execution will switch back to the threadpool at the first await)

Reply:
Should `InternalStartProcessWithKeepAlive()` be `async` at all? All the `await` are on `Task.Delay()` only.

Reply:
I reverted the change in 2027883